vpr/src/place/place_util.h

```diff
@@ -221,4 +222,15 @@ void alloc_and_load_legal_placement_locations(std::vector<std::vector<std::vecto
 
 ///@brief Performs error checking to see if location is legal for block type, and sets the location and grid usage of the block if it is legal.
 void set_block_location(ClusterBlockId blk_id, const t_pl_loc& location);
+
+/// @brief check if a specified location is within the device grid
+inline bool is_loc_on_chip(int x, int y) {
+    auto& device_ctx = g_vpr_ctx.device();
+    //return false if the location is not within the chip
+    return (x >= 0 && x < int(device_ctx.grid.width()) && y >= 0 && y < int(device_ctx.grid.height()));
+}
+
+/// @brief  Checks that each macro member location is legal based on the head position and its offset
```

Comment:
Remove enable_analytic_placer (see earlier comment).

It's also not included in the comment here, and every parameter should be described; the fact that describing it seems odd here is a hint that it's not a great parameter to have.